File: myTGbot.js
export default {
    async fetch(request, env, ctx) {
        if (request.method === "POST") {
            const payload = await request.json();
            let response;
            if ('message' in payload && 'text' in payload.message) {
                const chatId = payload.message.chat.id;
                const input = payload.message.text.trim().toLowerCase();
                const user_firstname = payload.message.from.first_name;

                switch (input) {
                    case '/start':
                        const startKeyboard = {
                            keyboard: [
                                [{ text: 'Menu' }]
                            ],
                            resize_keyboard: true
                        };
                        response = `Hello, ${user_firstname}! Welcome to Daniel's Bot! Use /help to explore the bot.`;
                        await this.sendMessageWithCustomKeyboard(env.API_KEY, chatId, response, startKeyboard);
                        return new Response('OK');

                    case '/help':
                        // Provide help information
                        response = `Here are some available commands:\n\n` +
                                   `   /start - Start the bot\n` +
                                   `   /menu - Display the menu\n` +
                                   `   /chatwithdaniel - Leave a message for Daniel\n` +
                                   `   /aboutdaniel - Learn more about Daniel\n` +
                                   `   /help - Display this help message`;
                        await this.sendMessage(env.API_KEY, chatId, response);
                        return new Response('OK');

                    case 'menu':
                        const menuKeyboard = {
                            inline_keyboard: [
                                [{ text: 'Leave Message to Daniel', callback_data: 'chat' }],
                                [{ text: 'Wanna Know About Daniel', callback_data: 'about' }]
                            ]
                        };
                        response = `Choose an option from the menu to proceed further.`;
                        await this.sendMessageWithInlineKeyboard(env.API_KEY, chatId, response, menuKeyboard);
                        return new Response('OK');

                    case 'chat with daniel':
                        // Provide a professional link to the Telegram user @dgety
                        response = `Use https://t.me/dgety to leave a message to Daniel.`;
                        await this.sendMessage(env.API_KEY, chatId, response);
                        return new Response('OK');

                    case 'about daniel':
                        // Provide information about Daniel
                        response = `Daniel is a software engineer.`;
                        await this.sendMessage(env.API_KEY, chatId, response);
                        return new Response('OK');

                    default:
                        // Handle other commands or text input here
                        response = `Sorry, I don't understand that command. Click 'Menu' to see more options.`;
                        await this.sendMessage(env.API_KEY, chatId, response);
                        return new Response('OK');
                }
            } else if ('callback_query' in payload) {
                // Handle callback query from inline keyboard
                const chatId = payload.callback_query.message.chat.id;
                const data = payload.callback_query.data;

                if (data === 'chat') {
                    // Provide a professional link to the Telegram user @dgety
                    response = `Use https://t.me/dgety to leave message to Daniel.`;
                } else if (data === 'about') {
                    // Provide information about Daniel
                    response = `Daniel is a software engineer.`;
                }

                await this.sendMessage(env.API_KEY, chatId, response);
                return new Response('OK');
            }
        }
        return new Response('OK');
    },

    async sendMessage(apiKey, chatId, text) {
        const url = `https://api.telegram.org/bot${apiKey}/sendMessage?chat_id=${chatId}&text=${text}`;
        const data = await fetch(url).then(resp => resp.json());
    },

    async sendMessageWithCustomKeyboard(apiKey, chatId, text, keyboard) {
        const url = `https://api.telegram.org/bot${apiKey}/sendMessage`;
        const body = JSON.stringify({
            chat_id: chatId,
            text: text,
            reply_markup: JSON.stringify(keyboard),
        });

        const headers = {
            'Content-Type': 'application/json',
        };

        const data = await fetch(url, { method: 'POST', headers, body }).then(resp => resp.json());
    },

    async sendMessageWithInlineKeyboard(apiKey, chatId, text, keyboard) {
        const url = `https://api.telegram.org/bot${apiKey}/sendMessage`;
        const body = JSON.stringify({
            chat_id: chatId,
            text: text,
            reply_markup: keyboard,
        });

        const headers = {
            'Content-Type': 'application/json',
        };

        const data = await fetch(url, { method: 'POST', headers, body }).then(resp => resp.json());
    }
};

File: myTGbot.py
from telegram import Update, ReplyKeyboardMarkup, KeyboardButton, InlineKeyboardButton, InlineKeyboardMarkup
from telegram.ext import Updater, CommandHandler, MessageHandler, Filters, CallbackContext, CallbackQueryHandler

TOKEN = "6325001638:AAGsGO6U5ePRhNxlRENJreiH8TFCznhwSvk"

def start(update: Update, context: CallbackContext) -> None:
    user_id = update.effective_user.id
    context.user_data.setdefault(user_id, {})

    update.message.reply_text("Hello! I am DanielBot. Type /help for a list of commands.",
                              reply_markup=get_menu_keyboard())

def help_command(update: Update, context: CallbackContext) -> None:
    update.message.reply_text("You can use the following commands:\n\n"
                              "/help - Display this help message\n"
                              "/start - Start the bot",
                              reply_markup=get_menu_keyboard())

def menu(update: Update, context: CallbackContext) -> None:
    update.message.reply_text("For further assistance, please select one of the options below:", reply_markup=get_inline_menu_keyboard())

def about_daniel(update: Update, context: CallbackContext) -> None:
    about_daniel_text = (
        "Daniel is a software engineer. He graduated from Bahir Dar University with a BSc in Software Engineering. "
        "He has backend skills and is certified in backend web development from ALX."
    )
    # Check if the message content or reply markup has changed
    if (update.callback_query.message.text != about_daniel_text or
            update.callback_query.message.reply_markup != get_inline_menu_keyboard()):
        update.callback_query.edit_message_text(about_daniel_text, reply_markup=get_inline_menu_keyboard())
    else:
        # If there's no change, do nothing to avoid the BadRequest error
        pass

def get_menu_keyboard():
    keyboard = [
        [KeyboardButton("Menu")],
    ]
    return ReplyKeyboardMarkup(keyboard, resize_keyboard=True)

def get_inline_menu_keyboard():
    keyboard = [
        [InlineKeyboardButton("Chat with Daniel", url="https://t.me/dgety")],
        [InlineKeyboardButton("About Daniel", callback_data="about_daniel")],
    ]
    return InlineKeyboardMarkup(keyboard, resize_keyboard=True)

def button_callback(update: Update, context: CallbackContext) -> None:
    query = update.callback_query
    query.answer()

    if query.data == "about_daniel":
        about_daniel(update, context)

def main() -> None:
    updater = Updater(TOKEN)
    dispatcher = updater.dispatcher

    dispatcher.add_handler(CommandHandler("start", start))
    dispatcher.add_handler(CommandHandler("help", help_command))
    dispatcher.add_handler(MessageHandler(Filters.regex('^(Menu)$'), menu))
    dispatcher.add_handler(CallbackQueryHandler(button_callback))

    updater.start_polling()
    updater.idle()

if __name__ == '__main__':
    main()
